src/swap/defi/thorchain.ts

```diff
@@ -438,6 +438,10 @@ export function makeThorchainPlugin(
     }
     const { fromNativeAmount, toNativeAmount, limit } = calcResponse
 
+    if (lte(limit, '0')) {
+      throw new Error('Swap would produce negative return')
```

Comment:
Returning a "below limit" error would be a better UX, but I have no idea how to calculate the parameters for that error, so this is fine. This will be the lowest priority message, since it isn't one of the known types, but at least we don't give a bogus quote.

Reply:
Yes, that's what I wanted, but I have no way of calculating it either.